### PR TITLE
docs: Add some info about automation testing

### PR DESCRIPTION
**Pull request by radeva**
Add section for automation testing with Appium and refer to Appium and the plugin itself

### CODE REVIEW

**Review by NickIliev — APPROVED** [review]

**Review by SvetoslavTsenov — APPROVED** [review]

File: docs/tooling/testing/end-to-end-testing/automation-testing.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+title: End to End (E2E) Testing
+description: Write and execute UI E2E automation tests to ensure that newly added features are working correctly and no regressions are introduced in the mobile app.
+position: 10
+tags: ui testing, app ui testing, nativescript ui testing, automation testing, app automation testing, nativescript automation testing, appium, ui test automation, e2e testing
+slug: automation-testing
```
Comment (etabakov) [inline docs/tooling/testing/end-to-end-testing/automation-testing.md]: Let's change this to e2e-testing to match the title and be similar to the url.

```suggestion
slug: e2e-testing
```

File: docs/tooling/testing/end-to-end-testing/automation-testing.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+previous_url: /testing,/core-concepts/testing
+---
+
+# End to End (E2E) Testing
+
+While [Unit Testing]({% slug unit-testing %}) is used to test an isolated piece of code usually in a mocked environment, E2E testing allows to test your application workflows and make sure all the integration points are working as expected. You can literally test any screen and any workflow of your app.
```
Comment (etabakov) [inline docs/tooling/testing/end-to-end-testing/automation-testing.md]: Let's inverse the order here and introduce what is E2E testing first and after that compare it with unit testing. Currently, this looks like a continuation of another article which the user might not have read previously.

File: docs/tooling/testing/end-to-end-testing/automation-testing.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            ├── sample.e2e-test.ts
+            ├── setup.ts
+            ├── tsconfig.json
+        ├── ...
+        ├── package.json
+        ├── tsconfig.json
+
+> To avoid any incompatibilities between the source of *e2e* tests (ES6) and the source of the application (ES5), we recommend to exclude the *e2e* folder from the application's *tsconfig.json* file: `exclude": [ "e2e" ]`.
+
+### Files Preview
```
Comment (etabakov) [inline docs/tooling/testing/end-to-end-testing/automation-testing.md]:
```suggestion
### Folder Structure
```

**Review by etabakov — COMMENTED** [review]
Looks good so far

**Comment by vchimev** [comment]
Is this still work in progress?

**Comment by etabakov** [comment]
It is. Some more details are needed and those are marked with TODOs in the articles.

**Comment by etabakov** [comment]
@cla-bot check

**Comment by etabakov** [comment]
@cla-bot check

**Pull request merged**